certora/harness/OperatorsRegistryV1Harness.sol

```diff
@@ -0,0 +1,168 @@
+        for (; valIndex < op.keys;) 
+        {
+            (bytes memory valData) = ValidatorKeys.getRaw(opIndex, valIndex);
+            if (validatorDataHash == keccak256(abi.encodePacked(valData)))  //element found
+            {
+                if (valIndex < op.requestedExits) return 4; //exited
+                if (valIndex < op.funded) return 3; //funded
+                if (valIndex < op.limit) return 2; //fundable
+                return 1; //available
+            }
+        }
+        return 0;   //not present in the list
+    }
+
+    //Returns current state of given validator with respect to given operator
+    //0 = not present
+    //1 = available = present but not fundable
+    //2 = fundable
+    //3 = funded, not exited
+    //4 = exited (funded in the past)
+    function getValidatorStateByIndex(uint256 opIndex, uint256 valIndex) external view returns (uint256) 
+    {
+        OperatorsV2.Operator memory op = OperatorsV2.get(opIndex);
+        if (valIndex < op.requestedExits) return 4; //exited
+        if (valIndex < op.funded) return 3; //funded
+        if (valIndex < op.limit) return 2; //fundable
+        if (valIndex < op.keys) return 1; //available
+        return 0;   //not present in the list
+    }
+
+    function operatorStateIsValid(uint256 opIndex) external view returns (bool) {
+        if (opIndex >= OperatorsV2.getAll().length) return false;
+        OperatorsV2.Operator memory op = OperatorsV2.get(opIndex);
+        return op.keys >= op.limit &&
+            op.limit >= op.funded &&
+            op.funded >= op.requestedExits;
+    }
+
+    function operatorStateIsValid_cond1(uint256 opIndex) external view returns (bool) {
+        OperatorsV2.Operator memory op = OperatorsV2.get(opIndex);
+        return op.keys >= op.limit;
+    }
+
+    function operatorStateIsValid_cond2(uint256 opIndex) external view returns (bool) {
+        OperatorsV2.Operator memory op = OperatorsV2.get(opIndex);
+        return op.limit >= op.funded;
+    }
+
+    function operatorStateIsValid_cond3(uint256 opIndex) external view returns (bool) {
+        OperatorsV2.Operator memory op = OperatorsV2.get(opIndex);
+        return op.funded >= op.requestedExits;
+    }
+
+    function getOperatorState(uint256 opIndex) external view 
+        returns (uint256, uint256, uint256, uint256, uint32, bool, address) {
+        OperatorsV2.Operator memory op = OperatorsV2.get(opIndex);
+        uint32 stoppedCount = _getStoppedValidatorsCount(opIndex);
+        return (op.keys, op.limit, op.funded, op.requestedExits, 
+            stoppedCount, op.active, op.operator);
+    }
+
+    function operatorIsActive(uint256 opIndex) external view returns (bool) {
+        OperatorsV2.Operator memory op = OperatorsV2.get(opIndex);
+        return op.active;
+    }
+
+    function getValidatorKey(uint256 opIndex, uint256 valIndex) external view returns (bytes memory) {
```

Comment:
What if the valIndex is out-of-bounds? It will revert, right?
Better return an empty bytes in that case.